packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,225 @@
+import {focusRing, StyleProps, getAllowedOverrides} from './style-utils' with {type: 'macro'};
+import {style} from '../style/spectrum-theme' with {type: 'macro'};
+import {centerBaseline} from './CenterBaseline';
+import {IconContext} from './Icon';
+import {raw} from '../style/style-macro' with {type: 'macro'};
+
+
+interface SegmentedControlProps extends ValueBase<string|null, string>, InputDOMProps, FocusEvents, StyleProps, AriaLabelingProps {
+  children?: ReactNode,
+  /**
+   * The style of the track.
+   * 
+   * @default 'none'
+   */
+  trackStyle?: 'none' | 'track',
+  isDisabled?: boolean
+}
+interface ControlItemProps extends Omit<RadioProps, 'children'>, StyleProps {
+  children?: ReactNode
+}
+
+const segmentedControl = style({
+  fontFamily: 'sans',
+  fontSize: 'control',
+  fontWeight: 'normal',
+  display: 'flex',
+  minWidth: {
+    trackStyle: {
+       // this is an arbitrary value. there is no minWidth in the tokens but the size of the each segment is dependent on the width of the segmented control so without it, it looks strange
+      track: 240
+    }
+  }
+}, getAllowedOverrides())
+
+const controlItem = style({
+  ...focusRing(),
+  display: 'flex',
+  gap: 'text-to-visual',
+  color: {
+    default: {
+      trackStyle: {
+        none: 'gray-600',
+        track: 'gray-700'
+      }
+    },
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    },
+  },
+  backgroundColor: {
+    trackStyle: {
+      none: {
+        isSelected: {
+          default: 'gray-100',
+          isHovered: 'gray-200',
+          isDisabled: 'disabled'
+        },
+        forcedColors: {
+          isSelected: 'Highlight',
+          isDisabled: {
+            isSelected: 'GrayText'
+          }
+        }
+      }, 
+      track: {
+        isSelected: {
+          default: 'gray-25'
+        },
+        forcedColors: {
+          isSelected: 'Highlight',
+          isDisabled: {
+            isSelected: 'GrayText'
+          },
+        }
+      }
+    },
+  },
+  // the padding should be a little less for segmented controls that only contain icons but not sure of a good way to do that
+  paddingX: {
+    trackStyle: {
+      none: 12
+    }
+  },
+  paddingY: 8,
+  alignItems: 'center',
+  boxSizing: 'border-box',
+  borderStyle: {
+    trackStyle: {
+      track: 'solid'
+    }
+  },
+  borderWidth: {
+    trackStyle: {
+      track: 2
+    },
+  },
+  borderColor: {
+    trackStyle: {
+      track: {
+        default: 'transparent',
+        isSelected: {
+          default: 'gray-900',
+          isDisabled: 'disabled'
+        },
+        forcedColors: {
+          isDisabled: 'GrayText',
+          isSelected: {
+            default: 'Highlight',
+            isDisabled: 'GrayText'
+          }
+        }
+      }
+    }
+  },
+  borderRadius: 'lg',
```

Comment:
looks like maybe it should be 'm'? or more probably, 'control'

Reply:
i can switch it to 'control' but it looks the same as 'lg' to me